win32ss/user/user32/windows/mdi.c

```diff
@@ -2047,6 +2047,28 @@ GetCascadeChildProc(HWND hwnd, LPARAM lParam)
     return TRUE;
 }
 
+static void
+QuerySizeFix(HWND hwnd, LPINT pcx, LPINT pcy)
+{
+    MINMAXINFO mmi;
+    DWORD_PTR dwResult;
+
+    mmi.ptMinTrackSize.x = mmi.ptMinTrackSize.y = 0;
+    mmi.ptMaxTrackSize.x = mmi.ptMaxTrackSize.y = MAXLONG;
+    if (SendMessageTimeoutW(hwnd, WM_GETMINMAXINFO, 0, (LPARAM)&mmi,
+                            SMTO_ABORTIFHUNG | SMTO_NORMAL, 120, &dwResult))
+    {
+        if (*pcx < mmi.ptMinTrackSize.x)
+            *pcx = mmi.ptMinTrackSize.x;
+        if (*pcy < mmi.ptMinTrackSize.y)
+            *pcy = mmi.ptMinTrackSize.y;
+        if (*pcx > mmi.ptMaxTrackSize.x)
+            *pcx = mmi.ptMaxTrackSize.x;
+        if (*pcy > mmi.ptMaxTrackSize.y)
+            *pcy = mmi.ptMaxTrackSize.y;
```

Comment:
You can use the min/max macros to simplify the code:
```c
*pcx = min(max(*pcx, mmi.ptMinTrackSize.x), mmi.ptMaxTrackSize.x);
*pcy = min(max(*pcy, mmi.ptMinTrackSize.y), mmi.ptMaxTrackSize.y);
```

Reply:
Ok.
■